docs: First page for new documentation

kostis-codefresh opened the pull request:
Ongoing work for https://github.com/argoproj/argo-cd/issues/5635
See how it looks at https://kostis-argo-cd.readthedocs.io/en/first-page/

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [X] Does this PR require documentation updates?
* [X] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

terrytangyuan commented on docs/index.md:
```diff
@@ -0,0 +1,89 @@
+## Community 
+
+* Visit the [Argo blog](https://blog.argoproj.io/)
+* Check the `#argo-cd` channel at the [CNCF slack](https://cloud-native.slack.com) 
```
Use this link instead? https://argoproj.github.io/community/join-slack/

terrytangyuan commented on docs/index.md:
```diff
@@ -0,0 +1,89 @@
+
+* Visit the [Argo blog](https://blog.argoproj.io/)
+* Check the `#argo-cd` channel at the [CNCF slack](https://cloud-native.slack.com) 
+* Visit the [Argo Subreddit](https://www.reddit.com/r/argoproj/)
```
Add link to Twitter and LinkedIn? Perhaps link to argoproj/argoproj instead.

terrytangyuan commented on docs/index.md:
```diff
@@ -0,0 +1,89 @@
+
+[![ArgoCD UI example](assets/argocd-ui-example.png)](https://cd.apps.argoproj.io/)
+
+Argo CD is one of the major tools in the [Argo Family of projects](https://argoproj.github.io/).
```
Perhaps “one of the core subproject of Argo”?

jannfis commented:
Just as a quick heads-up, previews can also be seen from Read The Doc's PR builds:

![image](https://user-images.githubusercontent.com/3942683/154220927-04039df3-b72c-495a-8952-19617b0ee23c.png)

jannfis commented on docs/index.md:
```diff
@@ -0,0 +1,89 @@
+* [Sync waves](syncing/waves.md) to control order of resource deployment
+* Audit trails for application events and API calls
+* [Prometheus metrics](operations/metrics.md)
+* Parameter overrides for overriding ksonnet/helm parameters in Git
```
```suggestion
* Parameter overrides for overriding Helm parameters in Git
```

jannfis commented on docs/index.md:
```diff
@@ -0,0 +1,89 @@
+
+* [kustomize](https://kustomize.io) applications
+* [helm](https://helm.sh) charts
+* [ksonnet](https://ksonnet.io) applications
```
```suggestion
```

I think we can get rid of ksonnet in the docs, because we are about to remove support for it (https://github.com/argoproj/argo-cd/pull/8479)

jannfis commented on docs/index.md:
```diff
@@ -0,0 +1,89 @@
+
+Organizations who have officially adopted Argo CD can be found [at the USERS file](https://github.com/argoproj/argo-cd/blob/master/USERS.md). Feel free to create a Pull Request and add your own organization if you are using Argo CD.  
+
+## Community 
```
I think the link to our GitHub repository should be here as well :)

jannfis commented on docs/index.md:
```diff
@@ -0,0 +1,89 @@
+## Feature highlight
+
+* Automated deployment of applications to specified target environments
+* Support for multiple config management/templating tools (Kustomize, Helm, Ksonnet, Jsonnet, plain-YAML)
```
```suggestion
* Support for multiple config management/templating tools (Kustomize, Helm, Jsonnet, plain-YAML)
```

jannfis commented on docs/index.md:
```diff
@@ -0,0 +1,89 @@
+
+## How it works
+
+Argo CD is a Kubernetes native application. You install it on a Kubernetes cluster as a controller. It them monitors a set of Git repositories and automatically creates/updates the Kubernetes Objects and resources described in Git.
```
```suggestion
Argo CD is a Kubernetes native application. You install it on a Kubernetes cluster as a controller. It then monitors a set of Git repositories and automatically creates/updates the Kubernetes Objects and resources described in Git.
```

jannfis reviewed (commented):
Thanks @kostis-codefresh, great work again! I love the new diagrams.

I only have a few nits, please see below.

kostis-codefresh commented:
@jannfis , @terrytangyuan I addressed all your comments.

Regarding the diagrams, I think that after we finish with the whole documentation, we can have a professional designer "beautify them"

jannfis commented:
I'm fine with the diagrams as-is, but yeah, we can of course revisit them later on :)

jannfis reviewed (approved):
LGTM
